docs/DESIGN_DESICIONS/applications_required_resource_access.md

```diff
@@ -0,0 +1,88 @@
+# Design Decision: Auto-Approving Application Roles in `required_resource_access`
+
+**Date:** March 2, 2025  
+**Author:** Sander Blomvågnes  
+
+## Context  
+
+Previously, when using `var.applications.app_name.required_resource_access`, only the `resource_app_id` needed to be provided. However, roles were not automatically assigned, requiring a Global Admin to manually grant the necessary permissions in Entra ID after deployment.  
+
+To enable automatic assignment of permissions, we now require the following details:  
```

Comment:
Instead of
> To enable automatic assignment of permissions, we now require the following details:

Can we write:
> Automatically grant admin consent for the application’s permissions within the organization by setting `auto_admin_consent = true` on `var.applications.app_name.required_resource_access`.